### DISCUSSION TITLE
Combat screen, obstacles' position on the battlefield

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#5667] [General]
In the original game, the position of obstacles on the battlefield is always the same according to the position of the square on the adventure map.
This means that if, in the same scenario, an attack takes place on the same square, the obstacles will be the same and their position will be the same, even if the scenario is played from the beginning again and again.

In fheroes2, the obstacles and their position are the same for the same square, but only for the same scenario session (for example, several attacks on the same square gives the same obstacles at the same positions).
If the scenario is replayed completely from the beginning, the obstacles are not the same nor their location.

In original game:
Completely new game.

![Same location combat obstacle H2 010](https://user-images.githubusercontent.com/43583994/179456332-d2dcce9c-6702-4dfe-9e74-b8181b07c164.png)

![Same location combat obstacle H2 001](https://user-images.githubusercontent.com/43583994/179456291-4c63244c-eeea-4c11-a2ed-2177791ae619.png)

![Same location combat obstacle H2 003](https://user-images.githubusercontent.com/43583994/179456307-723279ae-7ba5-4f16-ba41-9c614dd43395.png)

![Same location combat obstacle H2 005](https://user-images.githubusercontent.com/43583994/179456314-8a764aec-0196-4801-b1a1-7e95b085ad44.png)

![Same location combat obstacle H2 006](https://user-images.githubusercontent.com/43583994/179456321-abec694f-d4fe-4960-9cac-8e4a20a434ec.png)

In fheroes2:
Completely new game.

![Same location combat obstacle fh2 001](https://user-images.githubusercontent.com/43583994/179456461-70054985-fc56-4a3c-8ccf-fef97e3e01fa.png)

![Same location combat obstacle fh2 002](https://user-images.githubusercontent.com/43583994/179456469-50217c54-ab6e-4ba3-b4b9-3201cd82bf7c.png)

![Same location combat obstacle fh2 003](https://user-images.githubusercontent.com/43583994/179456485-074d2945-edc5-4044-a95f-1ae3a7015151.png)

![Same location combat obstacle fh2 004](https://user-images.githubusercontent.com/43583994/179456497-edfc1f32-8ed5-45ef-8a13-26c8b92b05b8.png)

![Same location combat obstacle fh2 005](https://user-images.githubusercontent.com/43583994/179456507-2b64d86f-8b49-4c6f-802f-f286285d62ae.png)

Should we reproduce the strictly identical operation of Heroes 2, or allow a difference in operation?
Should the difference in operation be a game option?

**Comment by Branikolog** [comment]
Hi, @LeHerosInconnu 
Various battlefields would add more replayability to all maps. I definitely prefer more variety. Also, map creator cannot affect spawn of battlefield objects so I don't feel it necessary to leave them on the same place every restart.

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

And for players who replay the same scenario (with as little randomness as possible) in order to achieve a better high score than the previous time, or a scenario specially created without randomness as part of a competition between players (this is possible with the way the original game works).
This can be a problem, as the arrangement of obstacles and their number can influence the course and outcome of a combat.

> Also, map creator cannot affect spawn of battlefield objects so I don't feel it necessary to leave them on the same place every restart.

Therefore, they are strictly identical if the player replays the same scenario again and again.

**Reply by oleg-derevenetz** [reply]
> And for players who replay the same scenario (with as little randomness as possible) in order to achieve a better high score than the previous time

Then why we have randomness at all? Let's throw it away and always see all the same monsters with all the same stack sizes in all the same places and go achieve "better high score than before". In other words, let's play chess :)

**Reply by LeHerosInconnu** [reply]
> > And for players who replay the same scenario (with as little randomness as possible) in order to achieve a better high score than the previous time
> 
> Then why we have randomness at all? Let's throw it away and always see all the same monsters with all the same stack sizes in all the same places and go achieve "better high score than before". In other words, let's play chess :)

Tell them. :rofl:

**Reply by Branikolog** [reply]
@LeHerosInconnu 
>And for players who replay the same scenario (with as little randomness as possible) in order to achieve a better high score than the previous time

I believe the times, when players replayed the same map to achieve high score, have passed away. Modern competitive gameplay mostly consists multiplayer battles, where placing obstacles on a battlefield in the same manner is not really necessary. 

>Therefore, they are strictly identical if the player replays the same scenario again and again.

Since map creator cannot place specified object on a needed place, it still appears randomly on creating the map. 

Btw, if we release good map generator players won't need to replay the same number of maps, creating dozens of new ones in a few clicks.

**Comment by LeHerosInconnu** [comment]
It's okay the way it works currently.
I close the discussion.
